src/autopas/selectors/AutoTuner.h

```diff
@@ -50,234 +34,175 @@ namespace autopas {
+    const size_t numIterationsForHomogeneity = 10;
+    const bool needsHomogeneityAndMaxDensity =
+        _tuningStrategy->smoothedHomogeneityAndMaxDensityNeeded() and
+        _iterationsSinceTuning > _tuningInterval - numIterationsForHomogeneity and
+        _iterationsSinceTuning <= _tuningInterval;
+
+    return {needsLiveInfo, needsHomogeneityAndMaxDensity};
+  }
+
+  /**
+   * Increase internal iteration counters by one.
    */
-  template <class PairwiseFunctor>
-  bool iteratePairwise(PairwiseFunctor *f, bool doListRebuild, std::vector<FullParticleCell<Particle>> &particleBuffers,
-                       std::vector<FullParticleCell<Particle>> &haloParticleBuffers);
+  void bumpIterationCounters();
 
   /**
-   * Returns whether the configuration will be changed in the next iteration.
-   * This does does not necessarily mean that the container will change.
+   * Returns whether rebuildNeighborLists() will be triggered in the next call to iteratePairwise().
+   * This might also indicate a container change.
    *
-   * @return True if the next iteratePairwise() call uses a different configuration. False otherwise.
+   * @return True if the the current iteration counters indicate a rebuild in the next iteration.
    */
-  bool willRebuild() {
-    if (_tuningStrategy->searchSpaceIsTrivial()) {
-      return false;
-    }
-    return _iterationsSinceTuning >= _tuningInterval and getCurrentNumSamples() >= _maxSamples;
-  }
+  bool willRebuildNeighborLists() const;
 
   /**
    * Get the currently selected configuration.
    * @return
    */
   [[nodiscard]] const Configuration &getCurrentConfig() const;
 
+  /**
+   * Ask the tuner for the next configuration to use.
+   * This either returns the already selected config or triggers a step of the tuning process.
+   * @return Tuple<Next configuration to use, still tuning>.
+   */
+  [[nodiscard]] std::tuple<Configuration, bool> getNextConfig();
+
+  /**
+   * Tell the tuner that the given config is not applicable.
+   *
+   * @note The applicability checking logic was moved out of the tuner because it needed the container,
+   * thus raising the compile complexity.
+   *
+   * @tparam PairwiseFunctor
+   * @param rejectedConfig
+   * @param indefinitely Whether the given config should be completely removed from the search space (aka rejected
+   * indefinitely).
+   * @return
+   */
+  [[nodiscard]] std::tuple<Configuration, bool> rejectConfig(const Configuration &rejectedConfig, bool indefinitely) {
+    // TODO: removing indefinitely will only be implemented once the AutoTuner manages the search space!
+    if (searchSpaceIsTrivial()) {
+      utils::ExceptionHandler::exception("Rejected the only configuration in the search space!\n{}",
+                                         rejectedConfig.toString());
+    }
+    const auto stillTuning = _tuningStrategy->tune(true);
+    return {getCurrentConfig(), stillTuning};
+  }
```

Comment:
You are right, this is not strictly necessary, however, I did it for multiple reasons:
1. This way you are more flexible and expressive about what you reject. For example, you can explicitly reject something and don't have to trick the tuner into first selecting it
2. This makes it easier to construct tests with rejected configurations.
3. You can reject multiple configurations directly after another.

So if you insist we can change the method (but then I would insist on renaming it to `rejectCurrentConfiguration()`) or we keep it as it is.

Reply:
Hmm, yea okay, we can leave it with the extra argument.
But maybe update the documentation a bit more, at least defining the return values.
Because I didn't expect a function called `rejectConfig()` to also get the next available config at first.